dags/oss_know/libs/zulip/crawl_stream.py

```diff
@@ -0,0 +1,64 @@
+import time
+import datetime
+from random import randint
+from dags.oss_know.libs.util.base import get_opensearch_client
+from oss_know.libs.util.log import logger
+from oss_know.libs.util.zulip import get_data_from_opensearch
+from oss_know.libs.base_dict.opensearch_index import OPENSEARCH_ZULIP_STREAM
+from oss_know.libs.util.opensearch_api import OpensearchAPI
+
+
+def crawl_zulip_stream(client, owner, repo, site, opensearch_conn_info):
+    max_try = 20
```

Comment:
There is a solid retry lib [tenacity](https://github.com/jd/tenacity) (already a dependency of the project), which can specify the retry times, exception types to catch, wait policy, etc. Here is [an example](https://github.com/oss-know/airflow-jobs/blob/development/dags/oss_know/libs/util/base.py#L53).